src/azure-cli/azure/cli/command_modules/sql/custom.py

```diff
@@ -2699,10 +2731,128 @@ def get_short_term_retention_mi(
     return policy
 
 
+def update_long_term_retention_mi(
+        cmd,
+        client,
+        database_name,
+        managed_instance_name,
+        resource_group_name,
+        weekly_retention=None,
+        monthly_retention=None,
+        yearly_retention=None,
+        week_of_year=None):
+    '''
+    Updates long term retention for managed database
+    '''
+    if not (weekly_retention or monthly_retention or yearly_retention):
+        raise CLIError('Please specify retention setting(s).')
+
+    if yearly_retention and not week_of_year:
+        raise CLIError('Please specify week of year for yearly retention.')
```

Comment:
We can also put these in a separate validator. Anyway, it is also ok to put it here.

Reply:
Whatever is preferred; I'm okay with either.